stratum/hal/lib/common/hal.cc

```diff
@@ -235,19 +285,37 @@ ::util::Status Hal::Run() {
     }
   }
 
+  // Clear any previous interrupts by locking semaphore
+  while (sem_trywait(&external_server_shutdown_sem) == 0)
+    ;
+  PCHECK(errno == EAGAIN);  // semaphore is locked
```

Comment:
Not actually sure that we want to do this. A client might have killed the process before we started, but it does replicate the previous behavior of not killing the server if the signal comes before we start the gRPC server.